### PR TITLE
Add Comms API `GET /commands` endpoint

### PR DESCRIPTION
**Pull request by GGP1**
|Related issue|
|---|
| Closes https://github.com/wazuh/wazuh/issues/24691 |

## Description

Implements the Communications API `GET /commands` endpoint.

Includes the API name modification: `Agent comms API` -> `Communications API`

## Tests

<details><summary>Index a new command</summary>

```console
root@wazuh-master:/# curl -X POST -H "Content-Type: application/json" -ku ***:*** http://wazuh-indexer:9200/commands/_doc -d '{"args": ["string"], "agent": {"id": "01912d07-b9b4-7528-bdad-15da902d651c"}, "status": "pending", "info": "string"}'
{"_index":"commands","_id":"Tx11VpEBYSr9jxqDIHBm","_version":1,"result":"created","_shards":{"total":2,"successful":1,"failed":0},"_seq_no":0,"_primary_term":1}
```

</details>

<details><summary>Get command</summary>

```console
root@wazuh-master:/# curl -H "Authorization: Bearer $TOKEN" -k https://0.0.0.0:27000/api/v1/commands
{"commands":[{"id":"UB2jVpEBYSr9jxqDgXAD","args":["string"],"agent":{"id":"01912d07-b9b4-7528-bdad-15da902d651c"},"status":"pending","info":"string"}]}
```

</details>

<details><summary>Check that its status was updated</summary>

```console
root@wazuh-master:/# curl -H "Content-Type: application/json" -ku ***:*** http://wazuh-indexer:9200/commands/_search -d '{"query": {"match": {"status": "sent"}}}'
{"took":0,"timed_out":false,"_shards":{"total":1,"successful":1,"skipped":0,"failed":0},"hits":{"total":{"value":1,"relation":"eq"},"max_score":0.6931471,"hits":[{"_index":"commands","_id":"UB2jVpEBYSr9jxqDgXAD","_score":0.6931471,"_source":{"args":["string"],"agent":{"id":"01912d07-b9b4-7528-bdad-15da902d651c"},"status":"sent","info":"string","id":"UB2jVpEBYSr9jxqDgXAD"}}]}}
```

</details>

<details><summary>Timeout error on no commands</summary>

```console
root@wazuh-master:/# curl -H "Authorization: Bearer $TOKEN" -k https://0.0.0.0:27000/api/v1/commands
{"message":"Request exceeded the processing time limit","code":408}
```

</details>

> I tested everything with multiple commands and the behaviour was the same.

<details><summary>Unit tests</summary>

```console
(venv) gasti@gasti:~/work/wazuh$ pytest apis/comms_api --disable-warnings
=========================================================== test session starts ============================================================
platform linux -- Python 3.10.14, pytest-7.3.1, pluggy-1.5.0
rootdir: /home/gasti/work/wazuh/apis/comms_api
plugins: asyncio-0.18.1, tavern-1.23.5, html-2.1.1, metadata-3.1.1, cov-4.1.0, anyio-4.1.0, trio-0.8.0, aiohttp-1.0.4
asyncio: mode=auto
collected 43 items                                                                                                                         

apis/comms_api/authentication/test/test_authentication.py ......                                                                     [ 13%]
apis/comms_api/core/test/test_commands.py .                                                                                          [ 16%]
apis/comms_api/core/test/test_events.py .                                                                                            [ 18%]
apis/comms_api/core/test/test_files.py ...                                                                                           [ 25%]
apis/comms_api/middlewares/test/test_logging.py ..                                                                                   [ 30%]
apis/comms_api/models/test/test_error.py ....                                                                                        [ 39%]
apis/comms_api/routers/test/test_authentication.py ....                                                                              [ 48%]
apis/comms_api/routers/test/test_commands.py ..                                                                                      [ 53%]
apis/comms_api/routers/test/test_events.py ..                                                                                        [ 58%]
apis/comms_api/routers/test/test_exceptions.py .....                                                                                 [ 69%]
apis/comms_api/routers/test/test_files.py ........                                                                                   [ 88%]
apis/comms_api/routers/test/test_utils.py .....                                                                                      [100%]

===================================================== 43 passed, 10 warnings in 1.82s ======================================================
```

</details>

### CODE REVIEW

File: framework/wazuh/core/indexer/base.py [hunk 1]
```diff
@@ -1,9 +1,27 @@
 import logging
+from enum import Enum
 from typing import Any, Dict, List, Tuple
 
 from opensearchpy import AsyncOpenSearch
 
 
+class Key(str, Enum):
```
Comment (GGP1) [inline framework/wazuh/core/indexer/base.py:8]: We should decide whether to use constants or an enum, I personally prefer the latter

File: apis/comms_api/authentication/test/test_authentication.py [hunk 2]
```diff
@@ -59,7 +60,7 @@ async def test_jwt_bearer_decode_ko(decode_token_mock):
     with pytest.raises(HTTPError) as exc:
         _ = await jwt(mock_req)
 
-    assert str(exc.value) == f'{status.HTTP_403_FORBIDDEN}: message'
+    assert str(exc.value) == '2706: Invalid authentication token'
```
Comment (nico-stefani) [inline apis/comms_api/authentication/test/test_authentication.py:63]: You can do this with the `match` argument of `pytest.raises`.

Something like:

```python
with pytest.raises(HTTPError, match='.* 2706.*') as exc:
        _ = await jwt(mock_req)
```

File: framework/wazuh/core/indexer/base.py (hunk 1 above)
Comment (nico-stefani) [reply framework/wazuh/core/indexer/base.py:8]: It's ok for me. I'd rename the class to something more specific, `QueryKey` or something like that.

**Review by nico-stefani — CHANGES_REQUESTED** [review]

File: framework/wazuh/core/indexer/base.py (hunk 1 above)
Comment (GGP1) [reply framework/wazuh/core/indexer/base.py:8]: The thing is that the enum contains keys for everything, queries, request bodies, responses, etc. I'll leave it as is for now, perhaps this class won't be imported anywhere except the helpers that we will develop (if necessary, we can modify it there, or create subclassses `QueryKey`, `ResponseKey`, `SearchKey`, etc.).

**Review by nico-stefani — APPROVED** [review]
LGTM

File: apis/comms_api/core/events.py [hunk 3]
```diff
@@ -16,4 +16,4 @@ async def post_stateful_events(events: Events) -> dict:
         Dictionary with the indexer response.
     """
     async with get_indexer_client() as indexer_client:
-        return await indexer_client.events.create(events)
+       return await indexer_client.events.create(events)
```
Comment (fdalmaup) [inline apis/comms_api/core/events.py:19]:
```suggestion
        return await indexer_client.events.create(events)
```

File: apis/comms_api/routers/events.py [hunk 4]
```diff
@@ -28,7 +28,7 @@ async def stateful_events(events: Events) -> JSONResponse:
         Indexer response.
     """
     try:
-        response = await post_stateful_events(events)
+        response =  await create_stateful_events(events)
```
Comment (fdalmaup) [inline apis/comms_api/routers/events.py:31]:
```suggestion
        response = await create_stateful_events(events)
```

File: framework/wazuh/core/indexer/base.py (hunk 1 above)
Comment (fdalmaup) [reply framework/wazuh/core/indexer/base.py:8]: I agree with Nico regarding a more expressive name like `IndexerKey` or similar so it gives a hint of what the class is referring to.

**Review by fdalmaup — CHANGES_REQUESTED** [review]

**Review by fdalmaup — APPROVED** [review]
LGTM! GJ :rocket:

**Pull request merged**